olive/passes/pytorch/awq.py

```diff
@@ -0,0 +1,224 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+from enum import Enum
+from typing import Any, Dict, Union
+
+import torch
+
+from olive.data.config import DataConfig
+from olive.hardware.accelerator import AcceleratorSpec, Device
+from olive.model import PyTorchModelHandler
+from olive.model.utils.path_utils import normalize_path_suffix
+from olive.passes import Pass
+from olive.passes.pass_config import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class AwqQuantizer(Pass):
+    """AWQ quantization."""
```

Comment:
Please rename the class to be AutoAWQQuantizer. There could be another implementation of AWQ.

Reply:
Done.